src/tools/genpolicy/Makefile

```diff
@@ -9,17 +9,37 @@ ifeq ($(ARCH), ppc64le)
      override ARCH = powerpc64le
  endif
 
+COMMIT_HASH := $(shell git rev-parse HEAD 2>/dev/null || true)
+# appends '-dirty' to the commit hash if there are uncommitted changes
+COMMIT_INFO := $(if $(shell git status --porcelain --untracked-files=no 2>/dev/null || true),${COMMIT_HASH}-dirty,${COMMIT_HASH})
+
+GENERATED_CODE = src/version.rs
+
+GENERATED_REPLACEMENTS= COMMIT_INFO
+GENERATED_FILES :=
+
+GENERATED_FILES += $(GENERATED_CODE)
+
+$(GENERATED_FILES): %: %.in
+	sed $(foreach r,$(GENERATED_REPLACEMENTS),-e 's|@$r@|$($r)|g') "$<" > "$@"
+
 .DEFAULT_GOAL := default
 default: build
 
-build:
-	@RUSTFLAGS="$(EXTRA_RUSTFLAGS) --deny warnings" cargo build --target $(TRIPLE) --$(BUILD_TYPE)
```

Comment:
This doesn't work in my machine. The error is coming from `--target $(TRIPLE)`. From https://doc.rust-lang.org/cargo/commands/cargo-build.html#compilation-options, target seems to be used to target different architectures. I don't think we need that for dev.

On the other hand, I do see how it's used in other makefiles like in https://github.com/microsoft/kata-containers/blob/msft-main/src/tools/kata-ctl/Makefile#L40, but I don't understand if it helps us. Maybe it's used in Kata CI? Found https://github.com/kata-containers/kata-containers/commit/201eec628a2dda2f92591eba1cb99c968ce819ea#diff-e157865d84d98087a8593904adb644a4d5348f59de4ca7111e9fedb9f406de70R97

Even if it is, we can leave this change in our fork and leave upstream makefile CI compatible

Reply:
I don't think we should change the behavior here as upstream builds using this target: https://github.com/kata-containers/kata-containers/blob/af3b19ed187e4412dee6ed96674714dd4d928422/tools/packaging/static-build/tools/build-static-tools.sh#L32

The problem is `$TRIPLE` gets evaluated to `x84_64-unknown-linux-musl` because `LIBC` defaults to musl in utils.mk. This works:

```
LIBC=gnu make
```